File: Code/1456B.cpp
#include <bits/stdc++.h>

using namespace std;

const int inf = 1000000000;

int main() {
	ios::sync_with_stdio(false);
	cin.tie(nullptr);

	int n;
	cin >> n;

	vector<int> a(n);
	for(int i = 0; i < n; ++i) {
		cin >> a[i];
	}

	vector<int> bit(31);
	for(int i = 0; i < n; ++i) {
		for(int j = 30; j >= 0; --j) {
			if(a[i] >> j & 1) {
				bit[j] += 1;
				break;
			}
		}	
	}

	if(*max_element(bit.begin(), bit.end()) > 2) {
		cout << 1 << '\n';
		exit(0);
	} 

	if(*max_element(bit.begin(), bit.end()) == 1) {
		cout << -1 << '\n';
		exit(0);
	}

	vector<int> pre(n + 1);
	for(int i = 0; i < n; ++i) {
		pre[i + 1] = pre[i] ^ a[i];
	}

	int ans = inf;
	for(int i = 1; i <= n; ++i) {
		for(int j = i; j <= n; ++j) {
			for(int k = j + 1; k <= n; ++k) {
				if((pre[j] ^ pre[i - 1]) > (pre[k] ^ pre[j])) {
					ans = min(ans, k - i - 1);
				}
			}
		}
	}

	cout << (ans == inf ? -1 : ans) << '\n';

	return 0;
}

File: Code/1456B.py
n = int(input())
a = list(map(int, input().split(' ')))
if (n > 60):
    print(1)
    exit(0)
pre = [None] * (n + 1)
pre[0] = 0
for i in range(n):
    pre[i + 1] = pre[i] ^ a[i]
ans = 1000000000
for i in range(1, n + 1):
    for j in range(i, n + 1):
        for k in range(j + 1, n + 1):
            if ((pre[i - 1] ^ pre[j]) > (pre[k] ^ pre[j])):
                ans = min(ans, k - i - 1)
if (ans == 1000000000):
    print(-1)
else:
    print(ans)

